modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java

```diff
@@ -135,44 +134,26 @@ public Event process(Event event) throws MuleException {
       ComponentIdentifier operationIdentifier =
           new ComponentIdentifier.Builder().withName(operationModel.getName())
               .withNamespace(extensionModel.getName().toLowerCase()).build();
-      Map<String, Object> originalParameterMap = this.resolverSet.resolve(event).asMap();
-      PolicyProvider policyProvider = policyManager.lookupPolicyProvider();
-      Optional<Policy> policy =
-          policyProvider.findOperationPolicy(event.getContext().getId(), operationIdentifier);
-      if (policy.isPresent()) {
-        Optional<PolicyOperationParametersTransformer> policyOperationParametersTransformer =
+      Optional<OperationPolicy> policy = policyManager.findOperationPolicy(event.getContext().getId(), operationIdentifier);
```

Comment:
Remember that policies are injected dynamically in the runtime. So it's not possible to use the start() method to fetch the policies. We know there's a performance concern here that will be takled by the gateway team. They know they will have to do effective catching of how they lookup for policies.